packages/design-system/src/components/Dialog/Dialog.tsx

```diff
@@ -334,86 +148,90 @@ export const Dialog = (props: DialogProps) => {
     };
   }, []);
 
-  const {
-    actions,
-    actionsClassName,
-    analytics,
-    ariaCloseLabel,
-    children,
-    className,
-    closeButtonSize,
-    closeButtonText,
-    closeButtonVariation,
-    closeIcon,
-    closeText,
-    escapeExits,
-    escapeExitDisabled,
-    headerClassName,
-    heading,
-    onExit,
-    size,
-    title,
-    ...modalProps
-  } = props;
+  // Toggle dialog open
+  const dialogRef = useRef(null);
+  const lastActiveElement = React.useRef(null);
+  const firstRender = React.useRef(true);
+
+  useLayoutEffect(() => {
+    if (window.HTMLDialogElement === undefined) {
+      dialogPolyfill.registerDialog(dialogRef.current);
+      // import('dialog-polyfill/dist/dialog-polyfill.css');
+    }
+  }, []);
+
+  useLayoutEffect(() => {
+    // prevents calling imperative methods on mount since the polyfill will throw an error since we are not using the `open` attribute
+    if (firstRender.current) {
```

Comment:
I had to add some extra `ref` hooks to address an IE bug where IE was expecting the `open` attribute to exist on the `<dialog>` element. Since this doesn't exist until after the modal's triggered, there was an issue.